_posts/2018-02-16-Java-Time2.md

```diff
@@ -0,0 +1,1564 @@
+        }
+        
+        return domain;
+    }
+    
+    public List<Domain> fetchAllDomainsForEnterprise(Enterprise enterprise) {
+        DomainsFetcher fetcher = enterprise.getDomains(); // 11
+        List<Domain> list = new ArrayList<>();
+
+        try {
+            if (fetcher.count() > 0) { // 12
+                list = fetcher.get(); // 13
+            }
+        } catch (RestException e) {
+            e.printStackTrace();
+        }
+
+        return list;
+    }
+    
+    public void deleteDomain(Domain domain) { // 14
+        if (domain == null) {
+            System.out.println("Domain does not exist");
+        } else {
+            System.out.println("Domain " + domain.getName() + " will be deleted");
+            try {
+                domain.delete(); // 15
+            } catch (RestException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+}
+{% endhighlight %}
+
+In Line 1 of the `verifyDomain` method, we have 3 arguments to supply the Enterprise parent object, related Domain Template object and the Network Descriptor holding the Domain name.
+
+Similar to other handlers, Line 2 determines whether there is an existing Domain object whose name matches the name that is passed in via `NetworkDescriptor`. If the Domain already exists, then we return that Domain object.  If the Domain does not exist, then in Line 3 we call the `createDomain` method to create and return a new Domain object.
+
+Line 4 of the `fetchDomain` method sets up a search filter based on the name of our domain.  In Line 5, we create an instance of `DomainsFetcher`.  Then Line 6 retrieves the Domain object matching the name in the search filter. 
+
+In Line 7 of the `createDomain` method, we call the constructor of the VSPK Domain class to create a brand new Domain object and then in Lines 8 and 9 we assign our domain name and ID of our existing domain template.  Finally, in Line 10, we attach our new Domain object to the already existing Enterprise object.
+
+Lines 11 to 15 of methods `fetchAllDomainsForEnterprise` and `deleteDomain` will be covered later when we discuss how to remove our network layout from the VSD.
+
+# Adding the Domain to our Network
+
+Here is the updated version of the `NetworkBuilder` class that integrates with `NetworkDomainHandler`.  There are 2 minor changes as shown below.  At Line 1, we instantiate the new handler and at Line 2, we call its `verifyDomain` method and pass it our existing Enterprise object, Domain Template object and the `NetworkDescriptor` object which contains the name of our domain.
+
+{% highlight java %}
+import java.util.List;
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.Domain;
+import net.nuagenetworks.vspk.v5_0.DomainTemplate;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443";

```

Comment:
Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`